src/munge.ts

```diff
@@ -69,50 +70,75 @@ export function removeCodec(sdpOrAv: Sdp | AvMediaDescription, codecName: string
 
 /**
  * Retain specific codecs, filtering out unwanted ones from the given SDP or audio/video media
- * description.
+ * description. The provided predicate should take in a single {@link codecInfo}, and only codecs
+ * for which the predicate returns true will be retained.
  *
- * Note: Done this way because of a feature not implemented in all browsers, currently missing in
- * Firefox. Once that is added we can use `RTPSender.getCapabilities` and filter those to call
- * with `RTCRtpTransceiver.setCodecPreferences` instead of doing this manually.
+ * Note: Done this way because of a feature that was only recently implemented in all browsers,
+ * previously missing in Firefox. You can also use `RTPSender.getCapabilities` and filter those to
+ * call with `RTCRtpTransceiver.setCodecPreferences` instead of doing this manually.
  *
  * @param sdpOrAv - The {@link Sdp} or {@link AvMediaDescription} from which to filter codecs.
- * @param allowedCodecNames - The names of the codecs that should remain in the SDP.
+ * @param predicate - A function used to determine which codecs should be retained.
+ * @returns A boolean that indicates if some codecs have been filtered out.
  */
 export function retainCodecs(
   sdpOrAv: Sdp | AvMediaDescription,
-  allowedCodecNames: Array<string>
-): void {
+  predicate: (codecInfo: CodecInfo) => boolean
+): boolean {
   const avMediaDescriptions = sdpOrAv instanceof Sdp ? sdpOrAv.avMedia : [sdpOrAv];
+  let filtered = false;
+
+  avMediaDescriptions.forEach((av) => {
+    av.codecs.forEach((codecInfo) => {
+      if (!predicate(codecInfo)) {
+        av.removePt(codecInfo.pt);
+        filtered = true;
+      }
+    });
+  });
+
+  return filtered;
+}
+
+/**
+ * Retain specific codecs, filtering out unwanted ones from the given SDP or audio/video media
+ * description by codec name.
+ *
+ * @param sdpOrAv - The {@link Sdp} or {@link AvMediaDescription} from which to filter codecs.
+ * @param allowedCodecNames - The names of the codecs that should remain in the SDP.
+ * @returns A boolean that indicates if some codecs have been filtered out.
+ */
+export function retainCodecsByCodecName(
+  sdpOrAv: Sdp | AvMediaDescription,
+  allowedCodecNames: Array<string>
+): boolean {
   const allowedLowerCase = allowedCodecNames.map((s) => s.toLowerCase());
 
-  avMediaDescriptions
-    .map((av) => {
-      return [...av.codecs.values()].map((c) => c.name as string);
-    })
-    .flat()
-    .filter((codecName) => !allowedLowerCase.includes(codecName.toLowerCase()))
-    .forEach((unwantedCodec) => removeCodec(sdpOrAv, unwantedCodec));
+  return retainCodecs(sdpOrAv, (codecInfo) =>
+    allowedLowerCase.includes(codecInfo.name?.toLowerCase() as string)
```

Comment:
How can you ensure that this function will always have codecInfo.name defined to cast type to string and not have some callback "value" for example?

Reply:
@bbaldino, do you remember why `CodecInfo.name` is optional? Is it just for convenience when building the `CodecInfo`, so you don't have to define a name until `CodecInfo.addLine()` is called?

Reply:
Discussed offline, but I think your memory is right: it's optional because we fill in that information later.  Bryce and I discussed some investigation of using an intermediary type while parsing that could then be finalized into another type which probably models required fields vs optional ones, so we can look into that.